### ISSUE TITLE
tr1/tests/thread is sporadically failing

### ISSUE THREAD

**Issue by StephanTLavavej** [microsoft/STL#1642]
#1627 encountered a sporadic failure in the legacy tr1 test suite. The failure log was:

```
Build setup steps:
Build steps:
Command: "C:\Program Files (x86)\Microsoft Visual Studio\2019\Preview\VC\Tools\MSVC\14.28.29812\bin\HostX64\x64\cl.EXE" "C:\a\1\s\tests\tr1\tests\thread\test.cpp" "-ID:\build\out\inc" "-IC:\a\1\s\tests\tr1\include" "-IC:\a\1\s\tests\std\include" "/nologo" "/Od" "/W4" "/w14061" "/w14242" "/w14265" "/w14582" "/w14583" "/w14587" "/w14588" "/w14749" "/w14841" "/w14842" "/w15038" "/w15214" "/w15215" "/w15216" "/w15217" "/sdl" "/WX" "/Zc:strictStrings" "/D_ENABLE_STL_INTERNAL_CHECK" "/bigobj" "/FIforce_include.hpp" "/D_ENFORCE_FACET_SPECIALIZATIONS=1" "/D_CRT_SECURE_NO_WARNINGS" "/Za" "/EHsc" "/MD" "/std:c++latest" "/permissive-" "-FeD:\build\tests\tr1\tests\thread\Output\14\thread.exe" "-link" "-LIBPATH:D:\build\out\lib\amd64" "-LIBPATH:C:\Program Files (x86)\Microsoft Visual Studio\2019\Preview\VC\Tools\MSVC\14.28.29812\lib\x64" "/MANIFEST:EMBED"
Exit Code: 0
Standard Output:
--
test.cpp

--

Intellisense response file steps:
Test setup steps:
Test steps:
Test step failed unexpectedly.
Command: "D:\build\tests\tr1\tests\thread\Output\14\thread.exe"
Exit Code: 1
Standard Output:
--
__STDCPP_THREADS__ = <1>

 GOT 0 != 1

 FAIL test 113 at line 336 in C:\a\1\s\tests\tr1\tests\thread\test.cpp: tgt <= STD chrono::system_clock::now() == true

FINISHED testing <thread>

***** 1 erroneous test cases in <thread> *****

***** 113 successful test cases in <thread> *****

#FAILED: <thread>

--
```

Code:

https://github.com/microsoft/STL/blob/47cd70315d560e4fe1ac699c0eb8ec8bfc5cb658/tests/tr1/tests/thread/test.cpp#L332-L342

Sporadic failures are doom. We should either fix this test to behave reliably, or skip it.

**Comment by CaseyCarter** [comment]
Is this not simply a consequence of #718?

**Comment by StephanTLavavej** [comment]
Failed again today, build 20210220.2.

**Comment by StephanTLavavej** [comment]
And again in 20210220.6. :pouting_cat:

**Comment by StephanTLavavej** [comment]
And again in 20210222.7. :scream_cat:

**Comment by StephanTLavavej** [comment]
The test also contains data races / timing assumptions:

https://github.com/microsoft/STL/blob/b52c3797e262cbce28166d92f9d1b8f2bfeb5ac5/tests/tr1/tests/thread/test.cpp#L16-L20

https://github.com/microsoft/STL/blob/b52c3797e262cbce28166d92f9d1b8f2bfeb5ac5/tests/tr1/tests/thread/test.cpp#L110-L113

https://github.com/microsoft/STL/blob/b52c3797e262cbce28166d92f9d1b8f2bfeb5ac5/tests/tr1/tests/thread/test.cpp#L136-L140

`called` is an ordinary `int` (not `atomic`, not protected by a lock), and this code assumes that the main thread can't possibly stall for 2 milliseconds between spawning `thr3` and verifying that `called` is still `0`.

**Comment by frederick-vs-ja** [comment]
Is this issue fixed by #4457?

**Comment by StephanTLavavej** [comment]
This was actually fixed by #2595 deleting the test completely.
